api/envoy/api/v2/route/route.proto

```diff
@@ -495,23 +515,23 @@ message RouteAction {
 
   // Specifies a set of headers that will be added to requests matching this
   // route. Headers specified at this level are applied before headers from the
-  // enclosing :ref:`envoy_api_msg_route.VirtualHost` and
+  // enclosing :ref:`envoy_api_msg_route.Route` and
   // :ref:`envoy_api_msg_RouteConfiguration`. For more information, including details on
   // header value syntax, see the documentation on :ref:`custom request headers
   // <config_http_conn_man_headers_custom_request_headers>`.
-  repeated core.HeaderValueOption request_headers_to_add = 12;
+  repeated core.HeaderValueOption request_headers_to_add = 12 [deprecated = true];
 
   // Specifies a set of headers that will be added to responses to requests
   // matching this route. Headers specified at this level are applied before
-  // headers from the enclosing :ref:`envoy_api_msg_route.VirtualHost` and
-  // :ref:`envoy_api_msg_RouteConfiguration`. For more information, including
-  // details on header value syntax, see the documentation on
+  // headers from the enclosing :ref:`envoy_api_msg_route.VirtualHost`,
+  // :ref:`envoy_api_msg_RouteConfiguration`, and :ref:`envoy_api_msg_route.Route`. For more
```

Comment:
Maybe put Route before VirtualHost (preserving the order in which they're applied)?